File: AspNetMvcUnleashed/0672329980_source/Chapter11/VB/MvcApplication1/MvcApplication1.Tests/Controllers/SimpleControllerTests.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Mvc

<TestClass()> _
Public Class SimpleControllerTests

    <TestMethod()> _
    Public Sub TimeIsCached()
        ' Arrange
        Dim timeMethod = GetType(SimpleController).GetMethod("Time")
        Dim outputCacheAttributes = timeMethod.GetCustomAttributes(GetType(OutputCacheAttribute), True)

        ' Assert
        Assert.IsTrue(outputCacheAttributes.Length > 0)
        For Each att As OutputCacheAttribute In outputCacheAttributes
            Assert.AreEqual(5, att.Duration)
        Next att
    End Sub

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter11/CS/MvcApplication1/MvcApplication1.Tests/Controllers/SimpleControllerTests.cs
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.Controllers;

namespace MvcApplication1.Tests.Controllers
{
    [TestClass]
    public class SimpleControllerTests
    {

        [TestMethod]
        public void TimeIsCached()
        {
            // Arrange
            var timeMethod = typeof(SimpleController).GetMethod("Time");
            var outputCacheAttributes = timeMethod.GetCustomAttributes(typeof(OutputCacheAttribute), true);

            // Assert
            Assert.IsTrue(outputCacheAttributes.Length > 0);
            foreach (OutputCacheAttribute att in outputCacheAttributes)
                Assert.AreEqual(5, att.Duration);
        }

    }
}
